pkg/datapath/linux/probes/probes.go

```diff
@@ -201,85 +208,101 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
 // returns an error when parameters required by Cilium are not enabled. It logs
 // warnings when optional parameters are not enabled.
 func (p *ProbeManager) SystemConfigProbes() error {
-	config := p.features.SystemConfig
-
 	if !p.KernelConfigAvailable() {
 		return ErrKernelConfigNotFound
 	}
-
-	// Required
-	if !config.ConfigBpf.Enabled() {
-		return fmt.Errorf("CONFIG_BPF kernel parameter is required")
-	}
-
-	if !config.ConfigBpfSyscall.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_SYSCALL kernel parameter is required")
-	}
-	if !config.ConfigNetSchIngress.Enabled() && !config.ConfigNetSchIngress.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_SCH_INGRESS kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsBpf.Enabled() && !config.ConfigNetClsBpf.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_BPF kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsAct.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_ACT kernel parameter is required")
-	}
-	if !config.ConfigBpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_JIT kernel parameter is required")
-	}
-	if !config.ConfigHaveEbpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_HAVE_EBPF_JIT kernel parameter is required")
-	}
-	// Optional
-	if !config.ConfigCgroupBpf.Enabled() {
-		log.Warning(
-			"CONFIG_CGROUP_BPF optional kernel parameter is not in kernel configuration")
-	}
-	if !config.ConfigLwtunnelBpf.Enabled() {
-		log.Warning(
-			"CONFIG_LWTUNNEL_BPF optional kernel parameter is not in kernel configuration")
+	requiredParams := p.GetRequiredConfig()
+	for param, kernelOption := range requiredParams {
+		if !kernelOption.Enabled {
+			module := ""
+			if kernelOption.CanBeModule {
+				module = " or module"
+			}
+			return fmt.Errorf("%s kernel parameter%s is required, Description:%s", param, module, kernelOption.Description)
```

Comment:
```suggestion
			return fmt.Errorf("%s kernel parameter%s is required, Description: %s", param, module, kernelOption.Description)
```
Missing space.

Thinking about it, we could maybe even change the wording, because “Description” is not very clear in the context. What about this?:
```golang
return fmt.Errorf("%s kernel parameter%s is required (needed for: %s)", param, module, kernelOption.Description)
```

Or maybe just skip the description altogether for required parameters (keep it just in the CLI?)
```golang
return fmt.Errorf("%s kernel parameter%s is required", param, module)
```
but no strong opinion, you can keep it if you like.